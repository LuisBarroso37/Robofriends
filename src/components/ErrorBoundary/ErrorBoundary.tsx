import React, { Component } from 'react';
import { IStateErrorBoundary } from '../../interfaces';

class ErrorBoundary extends Component<{}, IStateErrorBoundary> {
    constructor(props: never) {
        super(props);
        this.state = {
            hasError: false
        }
    };

    componentDidCatch(error: Error | null, info: object) {
        this.setState({
            hasError: true
        });
    };

    render() {
        if (this.state.hasError) {
            return <h1>Oooops. That is not good</h1>
        }

        return this.props.children
    };
};

export default ErrorBoundary;