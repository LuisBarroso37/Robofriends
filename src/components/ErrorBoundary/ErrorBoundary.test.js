import { shallow } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';
import React from 'react';

it('expect to render ErrorBoundary component', () => {
    expect(shallow(<ErrorBoundary />)).toMatchSnapshot();
});

it('expect hasError initial state to be false', () => {
    const wrapper = shallow(<ErrorBoundary />).instance();

    expect(wrapper.state.hasError).toEqual(false);
});

it('expect hasError state to be true if componentDidCatch is run', () => { 
    const wrapper = shallow(<ErrorBoundary />);
    wrapper.setState({ hasError: true });
    
    expect(wrapper).toMatchSnapshot();
});