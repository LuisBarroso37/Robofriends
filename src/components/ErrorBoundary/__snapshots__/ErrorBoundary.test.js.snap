// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect hasError state to be true if componentDidCatch is run 1`] = `ShallowWrapper {}`;

exports[`expect to render ErrorBoundary component 1`] = `ShallowWrapper {}`;
